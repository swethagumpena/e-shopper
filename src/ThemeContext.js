import React from 'react';

export const themes = {
  light: 'light',
  dark: 'dark',
};

const ThemeContext = React.createContext(themes.dark);
export default ThemeContext;

// const theme = {
//   light: {
//     backgroundColor: 'white',
//     color: 'black',
//   },
//   dark: {
//     backgroundColor: 'black',
//     color: 'white',
//   },
// };

// export const ThemeContext = React.createContext(theme.light);
