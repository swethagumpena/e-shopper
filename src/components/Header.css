.header{
    background-color: lightcoral;
    margin: auto;
    padding: 2em;
    display: flex;
}

