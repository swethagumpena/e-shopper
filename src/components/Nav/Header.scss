// .header{
//     display: flex;
//     justify-content: space-between;
// }
@import  '../../styles/index.scss';

.header-light{
<<<<<<< HEAD:src/components/Nav/Header.scss
    display: flex;
    justify-content: space-between;
    background-color:$primary-background-color-light;
    color:$primary-text-color-light;
=======
    display: flex;
    justify-content: space-between;
    background-color:$primary-background-color;
    color:black;
}

.header-dark{
    display: flex;
    justify-content: space-between;
    // background-color: grey;
    @include custom-mixin;
    color: white;
}

.header-dark a{
    color: white;
}

.header-dark a p{
    color: white;
>>>>>>> bc8dc9e8f7f0370b743c10d016316002aa8014ea:src/components/Nav/Header.css
}

.header-dark{
    display: flex;
    justify-content: space-between;
    // background-color: grey;
    background-color:$primary-background-color-dark;
    color:$primary-text-color-dark;
    // @include custom-mixin;
    // color: white;
}

.header-dark{
    a{
        color:$primary-text-color-dark;
        p {
            color:$primary-text-color-dark;
        }
    }
} 

// .header-dark a p{
//     color: white;
// }

.trolley-img  {
    float: left;
    width: 4rem;
}

.company {
    display: flex;
    padding: 1rem;
    p {
        font-size: larger;
        font-weight: bold;
        margin: 0;
<<<<<<< HEAD:src/components/Nav/Header.scss
=======
        color: green;
>>>>>>> bc8dc9e8f7f0370b743c10d016316002aa8014ea:src/components/Nav/Header.css
        padding-left: 1rem;
        display: flex;
        align-items: center;
    }
}

// .company p {
//     font-size: larger;
//     font-weight: bold;
//     margin: 0;
//     padding-left: 1rem;
//     display: flex;
//     align-items: center;
// }

.basket{
    display: flex;
}

.basket > a {
    text-decoration: none;
}

.basket p{
    margin: 0;
}

.orders{
    padding: 0.5rem;
    font-family:'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande', 'Lucida Sans', Arial, sans-serif;
    font-size: 1.5rem;
    color: black;
}

.basket-img{
    width: 3rem;
    padding: 1rem 0.5rem 1rem 1.5rem;
}

.cart {
    display: flex;
}

.order-count {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: flex-end;
}

.order-count p{
    margin-right: 1rem;
    color: black;
}  

.all-orders  {
    display: flex;
    align-items: center;
    font-family:'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande', 'Lucida Sans', Arial, sans-serif;
    font-size: 1.25rem;
    color: black;
}

@include media-mixin ($mobile-breakpoint){
    .company{
        padding: 1rem 0.5rem 1rem 0.5rem;
    }
    .trolley-img{
        width: 3rem;
    }
    .company p{
        font-size: medium;
        padding-left: 0.25rem;
    }
    .basket > a{
        font-size: medium;
    }
    hr{
        margin: 0 0 0.5rem 0;
    }
}