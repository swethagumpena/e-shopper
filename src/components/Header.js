import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './Header.css';

function Header(props) {
  const { cartCount } = props;
  return (
    <div className="header">
      <p>
        <Link to="/">logo</Link>
        {' '}
        E-Shopper
      </p>

      <Link to="/cart">
        <div className="cart">
          cart total :
          {' '}
          {cartCount}
        </div>
      </Link>
    </div>
  );
}

Header.propTypes = {
  cartCount: PropTypes.number.isRequired,
};

export default Header;
