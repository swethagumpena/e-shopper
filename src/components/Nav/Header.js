import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './Header.css';

function Header(props) {
  const { cartCount } = props;
  return (
    <>
      <div className="header">
        <div className="company">
          <Link to="/"><img className="trolley-img" src="assets/trolley.jpg" alt="logo" /></Link>
          {' '}
          <p>E-Shopper</p>
        </div>
        <div className="blank" />
        <div className="basket">
          <Link to="/orders" className="all-orders">
            {/* <div className="orders">All Orders</div> */}
            All Orders
          </Link>
          <Link to="/cart">
            <div className="cart">
              <img className="basket-img" src="assets/basket.jpeg" alt="basket" />
              <div className="order-count">
                <p>My Basket</p>
                <p>
                  {cartCount}
                  {' '}
                  items
                </p>
              </div>
            </div>
          </Link>
        </div>

      </div>
      <hr />
    </>
  );
}

Header.propTypes = {
  cartCount: PropTypes.number.isRequired,
};

export default Header;
