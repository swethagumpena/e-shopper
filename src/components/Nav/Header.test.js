import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import ThemeContext, { themes } from '../../ThemeContext';
import Header from './Header';

describe(Header, () => {
  test('should display E-Shopper', () => {
    render(<BrowserRouter><Header cartCount={2} /></BrowserRouter>);
    screen.getByText('E-Shopper');
  });

  test('should display All orders', () => {
    render(<BrowserRouter><Header cartCount={2} /></BrowserRouter>);
    screen.getByText('All Orders');
  });

  test('should display All orders', () => {
    render(<BrowserRouter><Header cartCount={2} /></BrowserRouter>);
    screen.getByText('My Basket');
  });

  test('should display All orders', () => {
    render(<BrowserRouter><Header cartCount={2} /></BrowserRouter>);
    screen.getByText('2 items');
  });

  test('should match snapshot for theme light', () => {
    const { container } = render(
      <BrowserRouter>
        <ThemeContext.Provider value={themes.light}>
          <Header cartCount={2} />
        </ThemeContext.Provider>
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });

  test('should match snapshot for theme dark', () => {
    const { container } = render(
      <BrowserRouter>
        <ThemeContext.Provider value={themes.dark}>
          <Header cartCount={2} />
        </ThemeContext.Provider>
      </BrowserRouter>,

    );
    expect(container).toMatchSnapshot();
  });

  test('should navigate to home page when E-shopper cart image is clicked', () => {
    render(<BrowserRouter><Header cartCount={2} /></BrowserRouter>);
    const homePageLink = screen.getByTestId('trolley-img');
    fireEvent.click(homePageLink);
    expect(document.location.href).toBe('http://localhost/');
  });

  test('should navigate to all orders page when All orders is clicked', () => {
    render(<BrowserRouter><Header cartCount={2} /></BrowserRouter>);
    const allOrdersPageLink = screen.getByText('All Orders');
    fireEvent.click(allOrdersPageLink);
    expect(document.location.href).toBe('http://localhost/orders');
  });

  test('should navigate to all cart page when My Basket is clicked', () => {
    render(<BrowserRouter><Header cartCount={2} /></BrowserRouter>);
    const cartPageLink = screen.getByText('My Basket');
    fireEvent.click(cartPageLink);
    expect(document.location.href).toBe('http://localhost/cart');
  });
});
