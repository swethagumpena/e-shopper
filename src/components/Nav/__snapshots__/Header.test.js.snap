// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot for theme dark 1`] = `
<div>
  <div
    class="header-dark"
    data-testid="header"
  >
    <div
      class="company"
    >
      <a
        href="/"
      >
        <img
          alt="logo"
          class="trolley-img"
          data-testid="trolley-img"
          src="assets/trolley.jpg"
        />
      </a>
       
      <p>
        E-Shopper
      </p>
    </div>
    <div
      class="basket"
      data-testid="All-Orders"
    >
      <a
        class="all-orders"
        href="/orders"
      >
        All Orders
      </a>
      <a
        href="/cart"
      >
        <div
          class="cart"
          data-testid="My-Basket"
        >
          <img
            alt="basket"
            class="basket-img"
            src="assets/basket.jpeg"
          />
          <div
            class="order-count"
          >
            <p>
              My Basket
            </p>
            <p>
              2 items
            </p>
          </div>
        </div>
      </a>
    </div>
  </div>
  <hr />
</div>
`;

exports[`Header should match snapshot for theme light 1`] = `
<div>
  <div
    class="header-light"
    data-testid="header"
  >
    <div
      class="company"
    >
      <a
        href="/"
      >
        <img
          alt="logo"
          class="trolley-img"
          data-testid="trolley-img"
          src="assets/trolley.jpg"
        />
      </a>
       
      <p>
        E-Shopper
      </p>
    </div>
    <div
      class="basket"
      data-testid="All-Orders"
    >
      <a
        class="all-orders"
        href="/orders"
      >
        All Orders
      </a>
      <a
        href="/cart"
      >
        <div
          class="cart"
          data-testid="My-Basket"
        >
          <img
            alt="basket"
            class="basket-img"
            src="assets/basket.jpeg"
          />
          <div
            class="order-count"
          >
            <p>
              My Basket
            </p>
            <p>
              2 items
            </p>
          </div>
        </div>
      </a>
    </div>
  </div>
  <hr />
</div>
`;
